sdk/storage/src/table/authorization/authorization_policy.rs

```diff
@@ -0,0 +1,94 @@
+use std::sync::Arc;
+
+use azure_core::{PipelineContext, PipelineError, Policy, PolicyResult, Request, Response};
+use http::{HeaderMap, HeaderValue, Method, Uri};
+use ring::hmac;
+
+use crate::{authorization::AuthorizationToken, table_context::TableContext};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AuthorizationPolicy {
```

Comment:
I'm unfortunately not familiar with authorization for the storage APIs. Can any of this be shared with the authorization implementation in Cosmos?

Reply:
I've never worked with Cosmos, so I don't want to say anything from basic reading. Of course, if there is a shared code we should use the same implementation

Reply:
Besides common bits, the auth scheme in Cosmos is a little different: there, for example, you can have "Users" with specific ACLs on documents[^cosmosuser]. Here we have SAS tokens[^sastokens]. 

I think It's best if every crate/function block keeps its own Auth policy.

[^cosmosuser]: https://docs.microsoft.com/en-us/rest/api/cosmos-db/create-a-user.
[^sastokens]: https://docs.microsoft.com/en-us/azure/storage/common/authorize-data-access.